src/Common/Tokenizer/ChineseTokenizer.cpp

```diff
@@ -0,0 +1,32 @@
+#include "ChineseTokenizer.h"
+
+#include <cassert>
+
+#include <base/defines.h>
+#include <cppjieba/Jieba.hpp>
+
+namespace DB
+{
+
+ChineseTokenizer & ChineseTokenizer::instance()
+{
+    static ChineseTokenizer tokenizer{};
+    return tokenizer;
+}
+
+std::vector<std::string> ChineseTokenizer::tokenize(const std::string & str)
+{
+#ifdef DEBUG_OR_SANITIZER_BUILD
+    assert(jieba_instance != nullptr);
+#endif
+    std::vector<std::string> tokens{};
+    jieba_instance->CutForSearch(str, tokens);
```

Comment:
Hmm, as far as I remembered Jieba accepts a "fine-granular" vs. "coarse-granular" mode parameter but I can't find it anywhere around here.

Reply:
There is no parameter for "fine-granular" and "coarse-granular" mode parameters. There are two different methods for that purpose: `jieba_instance->CutForSearch` and `jieba_instance->Cut`.

Reply:
Right ... I guess my question is if we could distinguish between fine- and coarse-granular at the SQL function level, i.e. add another parameter for users to specify either mode (or go with the default), and then internally call the right Jieba function.

Reply:
I removed the `tokenizeChinese` method and added a generic `tokenize` method to support multiple tokenizers: `default`, `ngram`, `noop` and `chinese` (enabled when ENABLE_CPPJIEBA is set to ON). I also added `fine-grained` and `coarse-grained` modes to the ChineseTokenizer. You can see the examples